src/_nebari/provider/cloud/amazon_web_services.py

```diff
@@ -121,6 +121,30 @@ def instances(region: str) -> Dict[str, str]:
     return {t: t for t in instance_types}
 
 
+@functools.lru_cache()
+def kms_key_arns(region: str) -> Dict[str, dict]:
+    """Return dict of available/enabled KMS key IDs and associated KeyMetadata for the AWS region."""
+    session = aws_session(region=region)
+    client = session.client("kms")
+    paginator = client.get_paginator("list_keys")
+    fields = [
+        "Arn",
+        "KeyUsage",
+        "KeySpec",
+        # "KeyState",
+        # "Origin",
+        # "KeyManager",
+        # "EncryptionAlgorithms",
+        # "MultiRegion",
+    ]
+    kms_keys = [
+        client.describe_key(KeyId=j["KeyId"]).get("KeyMetadata")
+        for i in paginator.paginate()
+        for j in i["Keys"]
+    ]
+    return {i["KeyId"]: {k: i[k] for k in fields} for i in kms_keys if i["Enabled"]}
```

Comment:
Could you add a comment at the beginning with a link to the docs for future reference? On the response object schema: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/describe_key.html

Reply:
Yes, I'll implement this in [PR#2788](https://github.com/nebari-dev/nebari/pull/2788)